pkg/deploy/deployment_che.go

```diff
@@ -112,10 +112,11 @@ func NewCheDeployment(cr *orgv1.CheCluster, cheImage string, cheTag string, cmRe
 										Scheme: corev1.URISchemeHTTP,
 									},
 								},
-								InitialDelaySeconds: 25,
-								FailureThreshold:    5,
+								InitialDelaySeconds: 60,
+								FailureThreshold:    10,
 								TimeoutSeconds:      5,
-							},
+								PeriodSeconds:       10,
+							}, // Will stop testing readiness after a minimum of 65 seconds and a maximum of 200 seconds (60 + 10*5 + 9*10)
```

Comment:
could you please clarify math?

> Will stop testing readiness after a minimum of 65 seconds

the readiness probe will continue to be called throughout the lifetime of the container, every periodSeconds, so that the container can make itself temporarily unavailable when one of its dependencies is unavailable, or while running a large batch job, performing maintenance, or something similar. [1] 

[1] https://blog.colinbreck.com/kubernetes-liveness-and-readiness-probes-how-to-avoid-shooting-y

Reply:
> could you please clarify math?

Sorry, it's a minimum of 60 seconds, not 65.

Here I'm considering the case of the POD start only:
- First, readiness probe will wait for 60 seconds after the start of the POD before checking anything
- Then the probe will start checking readiness. It might be that the first check will be successful immediately. In this case, POD will be seen as __*ready*__ after 60 seconds minimum. Else check will wait during a 5 seconds timeout.
- At most it will do it 10 times before declaring the POD as __*failed*__
- Between each check (10 checks at most, each of which can last 5 seconds at most), the probe will wait for 10 seconds (the period). That means that in our case this 10 seconds wait might occur at most 9 times (between the 10 checks).

So the math is:

Longest time (from POD start) after which the POD is declared __*ready*__ or __*failed*__ =
`InitialDelaySeconds` +
`FailureThreshold` * `TimeoutSeconds` +
(`FailureThreshold` - 1) * `PeriodSeconds`

Does it help .

> Will stop testing readiness after a minimum of 65 seconds

You're right, wording is misleading here: readiness probe wil continue its work. But at least the POD will, for the first time, be either declared __*ready*__ or __*failed*__ after this maximum delay. that's my main concern here.